### ISSUE TITLE
Generated code class name needs to take into account inner class's container class name

### ISSUE THREAD

**Issue by JerryShea** [OpenHFT/Chronicle-Wire#191]
If we have 2 inner classes (interfaces) named the same and create MethodWriters for each then GenerateMethodWriter will generate a class for whichever one comes first and try to use that same class for the second one. This may sound like a contrived example but it caused problems in some unit tests.
The generated class name for $package.$outerclass.$innerclass does not contain $outerclass

**Comment by dpisklov** [comment]
@JerryShea is there some context for this?

**Comment by RobAustin** [comment]
I have created the following failing test

```
package net.openhft.chronicle.wire;

import junit.framework.TestCase;
import net.openhft.chronicle.bytes.Bytes;

import static net.openhft.chronicle.wire.WireType.BINARY;

public class GenerateMethodWriterSameInterfaceDifferentOuterClassTest extends TestCase {

    public void test() {

        BINARY.apply(Bytes.elasticByteBuffer())
                .methodWriterBuilder(Outer1.InnerInterface.class)
                .updateInterceptor((methodName, t) -> true)
                .build()
                .x("hello world");

        BINARY.apply(Bytes.elasticByteBuffer())
                .methodWriterBuilder(Outer2.InnerInterface.class)
                .updateInterceptor((methodName, t) -> true)
                .build()
                .y("hello world");

    }


}

```

**Comment by RobAustin** [comment]
@JerryShea, Note: there are many other cases like this that we could come up with, like two classes with the same name but in different packages. We should ensure that we fix reasonable enhancements and not just some crazy theoretical use case someone constructed. My assumption is that this issue is a real problem, so we will fix it.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.30](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.30), [BOM-2.20.57](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.57)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
